ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/CommandRunnerTest.java

```diff
@@ -16,80 +16,90 @@
-    Deserializer<Command> commandDeserializer = new KafkaJsonDeserializer<>();
-    Serializer<CommandId> commandIdSerializer = new KafkaJsonSerializer<>();
-    Deserializer<CommandId> commandIdDeserializer = new KafkaJsonDeserializer<>();
+  private StatementExecutor getMockStatementExecutor() throws Exception {
+    CommandId topicCommandId =  new CommandId(CommandId.Type.TOPIC, "_CSASTopicGen");
+    CommandId csCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASStreamGen");
+    CommandId csasCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASGen");
+    CommandId ctasCommandId =  new CommandId(CommandId.Type.TABLE, "_CTASGen");
+    Map<CommandId, CommandStatus> statusStore = new HashMap<>();
+    statusStore.put(topicCommandId, new CommandStatus(CommandStatus.Status.SUCCESS, "Success"));
+    statusStore.put(csCommandId, new CommandStatus(CommandStatus.Status.SUCCESS, "Success"));
+    statusStore.put(csasCommandId, new CommandStatus(CommandStatus.Status.SUCCESS, "Success"));
+    statusStore.put(ctasCommandId, new CommandStatus(CommandStatus.Status.ERROR, "Error"));
 
-    KafkaConsumer<CommandId, Command> commandConsumer = new KafkaConsumer<>(
-        commandConsumerProperties,
-        commandIdDeserializer,
-        commandDeserializer
-    );
+    StatementExecutor statementExecutor = mock(StatementExecutor.class);
+    expect(statementExecutor.getStatuses()).andReturn(statusStore);
 
-    CommandRunner commandRunner = new CommandRunner(statementExecutor, new MockCommandStore
-        ("CT", commandConsumer, null,
-         new CommandIdAssigner(new MetaStoreImpl())));
-    return commandRunner;
+    return statementExecutor;
   }
 
-
   @Test
-  public void testThread() throws InterruptedException {
-    MockKsqkEngine mockKsqkEngine = new MockKsqkEngine(
-        TestUtils.getMockKsqlConfig(), new MockKafkaTopicClient());
-    StatementParser statementParser = new StatementParser(mockKsqkEngine);
-    StatementExecutor statementExecutor = new StatementExecutor(mockKsqkEngine, statementParser);
+  public void testNewCommandRun() throws Exception {
+    StatementExecutor statementExecutor = getMockStatementExecutor();
+    statementExecutor.handleStatement(anyObject(), anyObject());
+    expectLastCall().times(4);
+    replay(statementExecutor);
     CommandRunner commandRunner = getCommanRunner(statementExecutor);
-    new Thread(commandRunner).start();
-    Thread.sleep(50000);
-    commandRunner.close();
+    commandRunner.fetchAndRunCommands();
     CommandId topicCommandId =  new CommandId(CommandId.Type.TOPIC, "_CSASTopicGen");
     CommandId csCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASStreamGen");
     CommandId csasCommandId =  new CommandId(CommandId.Type.STREAM, "_CSASGen");
     CommandId ctasCommandId =  new CommandId(CommandId.Type.TABLE, "_CTASGen");
 
     Map<CommandId, CommandStatus> statusStore = statementExecutor.getStatuses();
```

Comment:
These next few lines aren't testing anything. You are testing that the `mock` returns the data you told it to. This doesn't belong as we are testing the `CommandRunner`.

Reply:
yes, refactored as I mentioned above.